diff --git a/libcxx/src/atomic.cpp b/libcxx/src/atomic.cpp
index b214ba1fd..43eb0ebc9 100644
--- a/src/atomic.cpp
+++ b/src/atomic.cpp
@@ -25,14 +25,14 @@
 #  if !defined(SYS_futex) && defined(SYS_futex_time64)
 #    define SYS_futex SYS_futex_time64
 #  endif
-#  define _LIBCPP_FUTEX(...) syscall(SYS_futex, __VA_ARGS__)
+#  define _LIBCPP_FUTEX(...) uk_syscall(SYS_futex, __VA_ARGS__)
 
 #elif defined(__FreeBSD__)
 
 #  include <sys/types.h>
 #  include <sys/umtx.h>
 
-#  define _LIBCPP_FUTEX(...) syscall(SYS_futex, __VA_ARGS__)
+#  define _LIBCPP_FUTEX(...) uk_syscall(SYS_futex, __VA_ARGS__)
 
 #elif defined(__OpenBSD__)
 
@@ -45,7 +45,7 @@
 
 // Baseline needs no new headers
 
-#  define _LIBCPP_FUTEX(...) syscall(SYS_futex, __VA_ARGS__)
+#  define _LIBCPP_FUTEX(...) uk_syscall(SYS_futex, __VA_ARGS__)
 
 #endif
 
